sql/integration.spec.ts

```diff
@@ -0,0 +1,269 @@
+import { expect } from "chai";
+import * as dfapi from "df/api";
+import * as dbadapters from "df/api/dbadapters";
+import { Sql } from "df/sql";
+import { build, ISelectOrBuilder, ISelectSchema } from "df/sql/builders/select";
+import { suite, test } from "df/testing";
+
+suite("builders", { parallel: true }, ({ before, after }) => {
+  let bigquery: dbadapters.IDbAdapter;
+  let snowflake: dbadapters.IDbAdapter;
+
+  before("create bigquery", async () => {
+    bigquery = await dbadapters.create(
+      dfapi.credentials.read("bigquery", "test_credentials/bigquery.json"),
+      "bigquery"
+    );
+  });
+
+  after("close bigquery", () => bigquery.close());
+
+  before("create snowflake", async () => {
+    snowflake = await dbadapters.create(
+      dfapi.credentials.read("snowflake", "test_credentials/snowflake.json"),
+      "snowflake"
+    );
+  });
+
+  after("close snowflake", () => snowflake.close());
+
+  const createSuite = (name: string, dbadapter: () => dbadapters.IDbAdapter, sql: Sql) => {
```

Comment:
nit: suggest a simple `for (const testCase of [{name: "...", dbadapter: ..., sql: ...}, {...}] { ... }` loop instead of this construction